openpype/hosts/nuke/plugins/publish/validate_rendered_frames.py

```diff
@@ -14,35 +14,44 @@ def get_instance(context, plugin):
         # Get the errored instances
         return get_errored_instances_from_context(context, plugin=plugin)
 
-    def repair_knob(self, instances, state):
+    def repair_knob(self, context, instances, state):
+        create_context = context.data["create_context"]
         for instance in instances:
-            node = instance.data["transientData"]["node"]
-            files_remove = [os.path.join(instance.data["outputDir"], f)
-                            for r in instance.data.get("representations", [])
-                            for f in r.get("files", [])
-                            ]
+            files_remove = [
+                os.path.join(instance.data["outputDir"], f_)
+                for r_ in instance.data.get("representations", [])
+                for f_ in r_.get("files", [])
+            ]
             self.log.info("Files to be removed: {}".format(files_remove))
-            for f in files_remove:
-                os.remove(f)
-                self.log.debug("removing file: {}".format(f))
-            node["render"].setValue(state)
+            for f_ in files_remove:
+                os.remove(f_)
+                self.log.debug("removing file: {}".format(f_))
+
+            # Reset the render knob
+            instance_id = instance.data["instance_id"]
+            created_instance = create_context.get_instance_by_id(
+                instance_id
+            )
+            created_instance.creator_attributes["render_target"] = state
+
             self.log.info("Rendering toggled to `{}`".format(state))
+        create_context.save_changes()
 
 
 class RepairCollectionActionToLocal(RepairActionBase):
     label = "Repair - rerender with \"Local\""
 
     def process(self, context, plugin):
         instances = self.get_instance(context, plugin)
```

Comment:
It's so weird that this method is called `get_instance` even though it returns a list.

Reply:
I'm not sure what exactly seems strange. I can only explain why I am using the `create_context.save_changes()` method since it is highlighted. The reason it needs to be saved is because the repair tool is modifying `creator_attributes` and those changes need to be saved back into the instance data knob at the instance node. This ensures that they are correctly invoked after a refresh action in the publisher.